hoomd/hpmc/update.py

```diff
@@ -472,6 +473,198 @@ def N(self):  # noqa: N802 - allow N as a function name
+            support 2 and 3.
+
+    Examples::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        mc.shape["A"] = dict(vertices=numpy.asarray([(1, 1, 1), (-1, -1, 1),
+                                                    (1, -1, -1), (-1, 1, -1)]) / 2)
+        vertex_move = hoomd.hpmc.shape_move.Vertex(stepsize={'A': 0.01},
+                                                   param_ratio=0.2,
+                                                   volume=1.0)
+        updater = hoomd.hpmc.update.Shape(shape_move=vertex_move,
+                                          trigger=hoomd.trigger.Periodic(1),
+                                          nselect=1,
+                                          nsweeps=1,
+                                          multi_phase=False,
+                                          num_phase=1)
+
+    Attributes:
+
+        shape_move (ShapeMove): Type of shape move to apply when updating shape
+            definitions
+
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        pretend (bool): When True the updater will not actually update the shape
+            definitions, instead moves will be proposed and the acceptance
+            statistics will be updated correctly.
+
+        nselect (int): Number of types to change every time the updater is
+            called.
+
+        nsweeps (int): Number of times to update shape definitions during each
+            triggered timesteps.
+
+        multi_phase (bool): When True MPI is enforced and shapes are updated
+            together for two boxes.
+
+        num_phase (int): How many boxes are simulated at the same time, now
+            support 2 and 3.
+    """
+
+    def __init__(self,
+                 shape_move,
+                 stepsize,
+                 trigger=1,
+                 pretend=False,
+                 nselect=1,
+                 nsweeps=1,
+                 multi_phase=False,
+                 num_phase=1):
+        super().__init__(trigger)
+        param_dict = ParameterDict(shape_move=hoomd.hpmc.shape_move.ShapeMove,
+                                   pretend=bool(pretend),
+                                   nselect=int(nselect),
+                                   nsweeps=int(nsweeps),
+                                   multi_phase=bool(multi_phase),
+                                   num_phase=int(num_phase))
+        param_dict["shape_move"] = shape_move
+        self._param_dict.update(param_dict)
+
+        # Set standard typeparameters for hpmc integrators
+        typeparam_stepsize = TypeParameter('stepsize',
+                                           type_kind='particle_types',
+                                           param_dict=TypeParameterDict(
+                                               float(stepsize), len_keys=1))
+
+        self._extend_typeparam([typeparam_stepsize])
+
+    def _add(self, sim):
+        if self.shape_move is not None:
+            self.shape_move._add(sim)
```

Comment:
We don't currently allow for `None` for `shape_move`.
```suggestion
        self.shape_move._add(sim)
```